lib/instrumentation/modules/nodemailer.js

```diff
@@ -0,0 +1,69 @@
+var shimmer = require('../shimmer')
+
+module.exports = function (nodemailer, agent, version, enabled) {
+  if (!enabled) return nodemailer
+  agent.logger.debug('shimming nodemailer.createTransport')
+  shimmer.wrap(nodemailer, 'createTransport', function (orig) {
+    return function () {
+      var transport = orig.apply(this, arguments)
+
+      shimTransport(transport)
+      return transport
+    }
+  })
+
+  function shimTransport (transport) {
+    agent.logger.debug('shimming transport.sendMail')
+
+    shimmer.wrap(transport, 'sendMail', function (orig) {
+      return function () {
+        const transaction = agent.currentTransaction ||
+                                        agent.startTransaction('Nodemailer', 'Nodemailer')
+
+        const span = transaction.startSpan('Send Email', 'nodemailer.sendMail')
```

Comment:
I don't think `agent.startTransaction(...)` should be used here. Transactions are meant to _contain_ a bunch of logically related things, like the async activity within HTTP requests or background jobs.

Rather than calling `agent.startTransaction(...)` and then `transaction.startSpan(...)`, it would be better to just use `agent.startSpan(...)`.

Reply:
I just wanted to catch every email sent to clients, if it's happening not under built-in transactions, then I auto create new one. But I think you are right, will fix it today.